GitUI/CommandsDialogs/FormPush.cs

```diff
@@ -121,7 +121,9 @@ void Init()
                 UserGitRemotes = _remotesManager.LoadRemotes(false).ToList();
                 BindRemotesDropDown(null);
 
-                UpdateBranchDropDown();
+                _NO_TRANSLATE_Branch.Text =
+                    _currentBranchName.IndexOfAny(['(', ' ', ')']) != -1 ? HeadText : _currentBranchName;
```

Comment:
Unfortunately, this has never worked and does not yet:

![image](https://github.com/gitextensions/gitextensions/assets/36601201/291250a7-1dea-4e6e-8113-62a6fae610ed)

Reply:
I was not really aware of this case and my changes on that was more code move (so without initial intention to fix it). Fixed now.